Input/Core/InputEventQueue.cs

```diff
@@ -56,5 +56,10 @@ public bool Dequeue(float targetTime, out InputEvent inputEvent)
 		}
 
 		#endregion
+
+		public void Sort()
+		{
+			m_List.Sort((a, b) => a.time.CompareTo(b.time));
```

Comment:
are we sure Sort doesn't allocate ?  i'm used to functions written like this allocating

Reply:
That's a good call--there's a lambda in there. But I think it wraps this in an IComparison object which means it only allocates once. In any case, it doesn't show up in the profiler.  With these changes I never see any per-frame allocations from the input prototype.

Reply:
There isn't any local capture occurring, just internal comparison of a/b, so perhaps no allocation beyond IComparison.  What about going with something like insertion or selection instead, for potentially better allocation (though of course with the caveat that list count would/may have an impact on sort time compared to list.sort(), which I thought used introsort)?

Reply:
There isn't any local capture occurring, just internal comparison of a/b, so perhaps no allocation beyond IComparison. What about going with something like insertion or selection instead, for potentially better allocation (though of course with the caveat that list count would/may have an impact on sort time compared to list.sort(), which I thought used introsort)?

Reply:
Well this is the strategy used by SortableList (I think. I just went with the most standard implementation, and I haven't seen any allocation (or particularly heavy sorting). Remember that this list (should) only include a handful of events, maybe 25-30, since we delta compress the on the input.